Sources/HummingbirdCore/Response/Response.swift

```diff
@@ -16,32 +16,34 @@ import HTTPTypes
 
 /// Holds all the required to generate a HTTP Response
 public struct Response: Sendable {
-    public var head: HTTPResponse
+    @inlinable
+    public var head: HTTPResponse {
+        get { HTTPResponse(status: status, headerFields: headers) }
+        set {
+            self.status = newValue.status
+            self.headers = newValue.headerFields
+        }
+    }
+
     public var body: ResponseBody {
         didSet {
             if let contentLength = body.contentLength {
-                self.head.headerFields[.contentLength] = String(describing: contentLength)
+                self.headers[.contentLength] = String(describing: contentLength)
             }
         }
     }
 
     public init(status: HTTPResponse.Status, headers: HTTPFields = .init(), body: ResponseBody = .init()) {
-        self.head = .init(status: status, headerFields: headers)
+        self.status = status
+        self.headers = headers
         self.body = body
-        if let contentLength = body.contentLength, headers[.contentLength] == nil {
-            self.head.headerFields[.contentLength] = String(describing: contentLength)
+        if let contentLength = body.contentLength, self.headers[.contentLength] == nil {
```

Comment:
`self.` was missing, leading to the original `headers` still being references. This lead to there being two active copies, which would cause a CoW